xrpl/transaction/types/credential_type.go

```diff
@@ -21,16 +33,75 @@ func (c *CredentialType) String() string {
 // - Not empty
 // - Valid hex string
 // - Length between MinCredentialTypeLength and MaxCredentialTypeLength
-func (c *CredentialType) IsValid() bool {
-	if c.String() == "" {
+func (c CredentialType) IsValid() bool {
+	hexStr := c.String()
+	if hexStr == "" {
 		return false
 	}
-
-	credTypeStr := c.String()
-	if !typecheck.IsHex(credTypeStr) {
+	if !typecheck.IsHex(hexStr) {
 		return false
 	}
-
-	length := len(credTypeStr)
+	length := len(hexStr)
 	return length >= MinCredentialTypeLength && length <= MaxCredentialTypeLength
 }
+
+// AuthorizeCredential represents an accepted credential for PermissionedDomainSet transactions.
+type AuthorizeCredential struct {
+	Issuer         Address
+	CredentialType CredentialType `json:"CredentialType"`
+}
+
+// Validate checks if the AuthorizeCredential is valid.
+func (a AuthorizeCredential) Validate() error {
+	if a.Issuer.String() == "" {
+		return ErrInvalidCredentialIssuer
+	}
+	if !a.CredentialType.IsValid() {
+		return ErrInvalidCredentialType
+	}
+	return nil
+}
+
+// Flatten returns a flattened map representation of the AuthorizeCredential.
+func (a AuthorizeCredential) Flatten() map[string]interface{} {
+	m := make(map[string]interface{})
+	if a.Issuer.String() != "" {
+		m["Issuer"] = a.Issuer.String()
+	}
+	if a.CredentialType != "" {
+		m["CredentialType"] = a.CredentialType.String()
+	}
+	return m
+}
+
+// validateCredentialsList validates a list of AuthorizeCredential objects.
+// It checks that:
+// - The credentials slice is not nil.
+// - The length is between 1 and maxCredentials.
+// - Each credential is valid.
+// - There are no duplicate credentials.
+func ValidateCredentialsList(credentials []AuthorizeCredential, maxCredentials int) error {
```

Comment:
This function should be a method of a `AuthorizeCredentialList` type. You can create one and add this method as `Validate`